src/njoy21/input/GROUPR.hpp

```diff
@@ -0,0 +1,67 @@
+class GROUPR {
+public:
+  /* nested types */
+  #include "njoy21/input/GROUPR/Card1.hpp"
+  #include "njoy21/input/GROUPR/Card2.hpp"
+  #include "njoy21/input/GROUPR/Card3.hpp"
+  #include "njoy21/input/GROUPR/Card4.hpp"
+  #include "njoy21/input/GROUPR/Card5.hpp"
+  #include "njoy21/input/GROUPR/Card6a.hpp"
+  #include "njoy21/input/GROUPR/Card6b.hpp"
+  #include "njoy21/input/GROUPR/Card7a.hpp"
+  #include "njoy21/input/GROUPR/Card7b.hpp"
+  #include "njoy21/input/GROUPR/Card8a.hpp"
+  #include "njoy21/input/GROUPR/Card8b.hpp"
+  #include "njoy21/input/GROUPR/Card8c.hpp"
+  #include "njoy21/input/GROUPR/Card8d.hpp"
+  #include "njoy21/input/GROUPR/Card9.hpp"
+  #include "njoy21/input/GROUPR/Card10.hpp"
+
+  #include "njoy21/input/GROUPR/src/readArbitraryNeutronStructure.hpp"
+  #include "njoy21/input/GROUPR/src/readArbitraryGammaStructure.hpp"
+  #include "njoy21/input/GROUPR/src/readFluxCalculatorParameters.hpp"
+  #include "njoy21/input/GROUPR/src/readTabulatedWeightFunction.hpp"
+  #include "njoy21/input/GROUPR/src/readAnalyticFluxParameters.hpp"
+  #include "njoy21/input/GROUPR/src/readInputResonanceFlux.hpp"
+  #include "njoy21/input/GROUPR/src/readCard9List.hpp"
+  #include "njoy21/input/GROUPR/src/readCard10List.hpp"
+
+  Card1 card1;
+  Card2 card2;
+  Card3 card3;
+  Card4 card4;
+  Card5 card5;
+  optional< std::pair< Card6a, Card6b > > arbitraryNeutronStructureCards;
+  optional< std::pair< Card7a, Card7b > > arbitraryGammaStructureCards;
+  optional< Card8a > card8a;
+  optional< Card8b > card8b;
```

Comment:
Not true
```cpp
GROUPR groupr(...);
REQUIRE( groupr.card8b );
REQUIRE( not groupr.card8c );
groupr.card8c = GROUPR::Card8c(...)
REQUIRE( groupr.card8c ); // uh oh
```

In order to prevent invalid state we can
1. encapsulation state and verify upon mutatation
1. use a data structure that doesn't admit invalid compositions

Module classes do not currently provide encapsulation. Adding verification would require quite a bit of hand written code on a module-by-module basis.